cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/mazha/CouchClient.java

```diff
@@ -465,13 +465,16 @@ public ChangesResult changes(PullFilter filter, Object since, Integer limit) {
     }
 
     public ChangesResult changes(String selector, Object since, Integer limit) {
-        Misc.checkNotNullOrEmpty(selector,null);
-        Map<String, Object> options = getParametrizedChangeFeedOptions(since, limit);
-        options.put("filter", "_selector");
-        URI changesFeedUri = uriHelper.changesUri(options);
-        HttpConnection connection = Http.POST(changesFeedUri, "application/json");
-        connection.setRequestBody(selector);
-        return executeToJsonObjectWithRetry(connection, ChangesResult.class);
+        Misc.checkNotNullOrEmpty(selector, null);
+        return changesRequestWithPost("_selector", selector, since, limit);
+    }
+
+    public ChangesResult changes(List<String> docIds, Object since, Integer limit) {
+        Misc.checkState((docIds != null && !docIds.isEmpty()), null);
+        Map<String, Object> docIdsMap = new HashMap<String, Object>();
+        docIdsMap.put("doc_ids", docIds);
+        String docsIdsDoc = JSONUtils.serializeAsString(docIdsMap);
+        return changesRequestWithPost("_doc_ids", docsIdsDoc, since, limit);
     }
 
     public ChangesResult changes(final Map<String, Object> options) {
```

Comment:
This version of `changes` should be made private and have its name changed to something like `changesRequestWithGet` to be consistent with its new counterpart `changesRequestWithPost`